test/ZNSRootRegistrar.test.ts

```diff
@@ -984,14 +984,48 @@ describe("ZNSRootRegistrar", () => {
       expect(finalstaked).to.equal(BigInt("0"));
       expect(finalToken).to.equal(ethers.ZeroAddress);
 
+      const protocolFee = getStakingOrProtocolFee(staked);
+
       // Verify final balances
-      const computedFinalBalance = balance + staked;
+      const computedFinalBalance = balance + staked - protocolFee;
       const finalBalance = await zns.meowToken.balanceOf(user.address);
       expect(computedFinalBalance).to.equal(finalBalance);
     });
   });
 
   describe("Revoking Domains", () => {
+    it("Sends a protocol fee as part of revoke flow", async () => {
+      await defaultRootRegistration({
+        user,
+        zns,
+        domainName: defaultDomain,
+        distrConfig: {
+          pricerContract: await zns.curvePricer.getAddress(),
+          paymentType: PaymentType.STAKE,
+          accessType: AccessType.OPEN,
+        },
+      });
+
+      const domainHash = await getDomainHashFromEvent({
+        zns,
+        user,
+      });
+
+      const [ price, stakeFee ] = await zns.curvePricer.getPriceAndFee(ethers.ZeroHash, defaultDomain, false);
+
+      const balanceBefore = await zns.meowToken.balanceOf(user.address);
+
+      // is revoke meant to be free if owner of parent? register subdomain is
+      await zns.rootRegistrar.connect(user).revokeDomain(domainHash);
+
+      const balanceAfter = await zns.meowToken.balanceOf(user.address);
+
+      // Stake fee and protocol fee are calculated the same way, so we can use
+      // stake fee here to represent that the difference in balance after revoking
+      // shows a protocol fee was sent as well

```

Comment:
here you already confused yourself. :) we are using the correct `protocolFee` here, since we've acquired if from zero hash. the thing that's wrong here is the naming.

Reply:
I feel like the comment is explanatory that this is the case and so it's not confusing in my opinion, but I removed this and made it more explicit using two await calls to get the fee properly